optaplanner-examples/src/main/java/org/optaplanner/examples/cheaptime/domain/solver/TaskAssignmentDifficultyComparator.java

```diff
@@ -19,22 +19,15 @@
 import java.io.Serializable;
 import java.util.Comparator;
 
-import org.apache.commons.lang3.builder.CompareToBuilder;
-import org.optaplanner.examples.cheaptime.domain.Task;
 import org.optaplanner.examples.cheaptime.domain.TaskAssignment;
 
 public class TaskAssignmentDifficultyComparator implements Comparator<TaskAssignment>, Serializable {
 
+    private static final Comparator<TaskAssignment> COMPARATOR = Comparator.comparing(TaskAssignment::getTask)
+            .thenComparingLong(TaskAssignment::getId);
+
     @Override
     public int compare(TaskAssignment a, TaskAssignment b) {
-        Task aTask = a.getTask();
-        Task bTask = b.getTask();
-        return new CompareToBuilder()
-                .append(aTask.getResourceUsageMultiplicand(), bTask.getResourceUsageMultiplicand())
```

Comment:
Looks like this changes the behavior? The COMPERATOR does use getResourceUsageMultiplicand()

Reply:
It does not change anything. It refers to the `Task`, which is now `Comparable` and does the rest of the comparison exactly as written here.

That said, I'll move that Comprator code here, to make it consistent with the other examples and to make it clearer what it actually does.